android/lib/src/main/java/org/vosk/android/SpeechService.java

```diff
@@ -258,9 +263,17 @@ public void run() {
 
                     //Checks if the queue is full
                     if (chunksInQueue >= RECOGNIZER_QUEUE) {
-                        listener.onTranscriptionFailed();
+                        transcriptionThread.isToResetRecognizer();
```

Comment:
We signal the transcriptionThread that we want to Reset (which clears the queue after the next iteration) but then if listener != null we clear the queue explicitly?

Reply:
That does not make sense. I will change it. Thanks